environs/bootstrap/bootstrap.go

```diff
@@ -144,6 +144,10 @@ type BootstrapParams struct {
 	// CAPrivateKey is the controller's CA certificate private key.
 	CAPrivateKey string
 
+	// ControllerServiceType is the service type to use for a k8s controller.
+	// Optional: only for k8s controller.
+	ControllerServiceType string
```

Comment:
We should be able to look this up from the cloud info on the server side.
I don't think there's a need to add a k8s specific attribute here

Reply:
as discussed, unfortunately we have to pass through the whole process.
Because k8s cloud type is only known by add-k8s cmd then Juju will lose this knowledge forever.